File: tests/benchmarks/resolution/fixtures/zig/validators.zig
const std = @import("std");

fn isNotEmpty(value: []const u8) bool {
    return value.len > 0;
}

pub fn validateName(name: []const u8) bool {
    if (!isNotEmpty(name)) return false;
    return name.len >= 2;
}

pub fn validateEmail(email: []const u8) bool {
    if (!isNotEmpty(email)) return false;
    for (email) |c| {
        if (c == '@') return true;
    }
    return false;
}

File: tests/benchmarks/resolution/fixtures/dart/validators.dart
bool isNotEmpty(String value) {
  return value.isNotEmpty;
}

bool validateName(String name) {
  if (!isNotEmpty(name)) return false;
  return name.length >= 2;
}

bool validateEmail(String email) {
  if (!isNotEmpty(email)) return false;
  return email.contains('@');
}

bool validateAmount(double amount) {
  return amount > 0;
}
